.github/workflows/gh-pages.yml

```
@@ -0,0 +1,40 @@
+name: Github Pages
+
+on:
+  push:
+    branches: [ "master" ]
+  workflow_dispatch:
+
+jobs:
+  gh-pages:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cachix/install-nix-action@v18
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          extra_nix_config: |
+            trusted-public-keys = hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= loony-tools:pr9m4BkM/5/eSTZlkQyRt57Jz7OMBxNSUiMC4FkcNfk=
+            substituters = https://cache.iog.io https://cache.nixos.org/ https://cache.zw3rk.com
+      - name: Build projects and haddocks
+        run: nix-shell --command "cabal update && cabal build --enable-documentation all && ./scripts/haddocks.sh"
+      - name: Add files
+        run: |
+              git config --local user.name ${{ github.actor }}
+              git config --local user.email "${{ github.actor }}@users.noreply.github.com"
+              cp -r ./haddocks/* ./
+              rm -rf haddocks
+              git add -A
```

Just in case `*.html` gets added to `.gitignore` in this repo as well, which is likely to happen sooner or later.
```suggestion
              git add -A --force
```